images.go

```diff
@@ -219,17 +219,44 @@ func (c *Client) CreateImageUpload(ctx context.Context, opts ImageCreateUploadOp
 
 // UploadImageToURL uploads the given image to the given upload URL.
 func (c *Client) UploadImageToURL(ctx context.Context, uploadURL string, image io.Reader) error {
-	// Linode-specific headers do not need to be sent to this endpoint
-	req := resty.New().SetDebug(c.resty.Debug).R().
-		SetContext(ctx).
-		SetContentLength(true).
-		SetHeader("Content-Type", "application/octet-stream").
-		SetBody(image)
+	clonedClient := *c.httpClient
+	clonedClient.Transport = http.DefaultTransport
 
-	_, err := coupleAPIErrors(req.
-		Put(uploadURL))
+	var contentLength int64 = -1
 
-	return err
+	if seeker, ok := image.(io.Seeker); ok {
+		size, err := seeker.Seek(0, io.SeekEnd)
+		if err != nil {
+			return err
+		}
+
+		_, err = seeker.Seek(0, io.SeekStart)
+		if err != nil {
+			return err
+		}
+
+		contentLength = size
+	}
```

Comment:
One more tiny thing, it looks like the client's User-Agent header isn't be respected here. Other than that this looks great!

Reply:
Good catch! This should be fixed now.